engine/c.cpp

```diff
@@ -50,8 +59,8 @@ struct KVDKSortedCollectionConfigs {
 };
 
 static char* CopyStringToChar(const std::string& str) {
-  char* result = reinterpret_cast<char*>(malloc(sizeof(char) * str.size()));
-  memcpy(result, str.data(), sizeof(char) * str.size());
+  char* result = static_cast<char*>(malloc(str.size()));
```

Comment:
why remove sizeof(char*)

Reply:
sizeof(char) is always 1 according to C99 standard.